docs/design/fabric-v2+/fpc-lifecycle-v2-new.puml

```diff
@@ -0,0 +1,139 @@
+entity "FPC_Peer2_Wrapper" as FPC_Peer2_Wrapper order 30
+participant Peer2 order 40
+end box
+
+box "Channel Abstraction"
+participant Channel order 70
+note over Channel
+    This box abstracts all other orgs and components.
+    Actions are supposed to be performed by channel participants.
+end note
+end box
+
+entity Orderer order 80
+
+note over Admin
+    Admin uses the fpc-peer wrappers, which abstract the Fabric Peer CLI.
+end note
+
+group chaincode package
+Admin -> FPC_Peer1_Wrapper : lifecycle chaincode package
+note right
+    **Chaincode package**
+    as tar.gz with
+    metadata.json = <path:(enclaves_path), type:(cc_lang), label:(cc_label), sgx_mode: (sim/hw)>
+    code.tar.gz : compressed file with enclave binary and mrenclave
+end note
+return
+end /' group chaincode package '/
+
+group install
+Admin -> FPC_Peer1_Wrapper ++: lifecycle chaincode install
+FPC_Peer1_Wrapper -> Peer1 ++: lifecycle chaincode install
+return packageId
+note over FPC_Peer1_Wrapper
+    Bookeeping operations to take into account that:
+    - FPC's initial version will run a chaincode on a designated peer
+    - the install lifecyle op is peer-specific
+    - the approveformyorg and commit ops are org-specific
+    So the approve stores data for an FPC chaincode installed locally,
+    and the commit checks for data of a locally-approved FPC chaincode.    
+end note
+FPC_Peer1_Wrapper -> FPC_Peer1_Wrapper : retrieve cc_language from package
+note over FPC_Peer1_Wrapper
+    Queryinstalled is redudant, yet consistent with implementation.    
+end note
+FPC_Peer1_Wrapper -> Peer1 ++: lifecycle chaincode queryinstalled
+return packageId
+FPC_Peer1_Wrapper -> FPC_Peer1_Wrapper : if cc_language==fpc then store packageId
+return packageId /' Admin -> FPC_Peer1_Wrapper '/
+end /' group install '/
+
+ref over Admin, FPC_Peer2_Wrapper, Peer2
+    install chaincode on Peer2
+end /' ref '/
+
+group /' approve '/
+Admin -> FPC_Peer1_Wrapper ++: lifecycle chaincode approveformyorg
+FPC_Peer1_Wrapper -> Peer1 ++: lifecycle chaincode approveformyorg
+note right
+    **Chaincode definition**
+    Name: cc_name, Version: mrenclave,
+    Sequence: sequence_number, Endorsement Policy: (2of3),
+    Plugins: <custom endorsement/validation plugins>
+end note
+Peer1 -> Orderer ++: approve transaction
+return transaction committed /' Peer1 -> Orderer '/
+return /' FPC_Peer1_Wrapper -> Peer1 '/
+note over FPC_Peer1_Wrapper
+    Bookkeping operations.
+    Note: packageId is an input parameter of approveformyorg
+end note
+FPC_Peer1_Wrapper -> FPC_Peer1_Wrapper : if packageId was stored then store cc_namie, cc_version
+return /' Admin -> FPC_Peer1_Wrapper '/
+
+Channel -> Channel : approvals from other parties
+note over Channel
+    Other participants must approve the chaincode definition,
+    and commit the corresponding transaction,
+    according to the LifecycleEndorsement policy.
+    So the party here has to loop to check for the approvals.
+end note
```

Comment:
I think this important step is not prominent enough .... I propose to remove the channel abstraction and add another ref box between approve and loop that refers to `approve` ... something similar like this
```
ref over Admin, FPC_Peer2_Wrapper, Peer2
    approvals form other orgs
end
```

and also including your note